File: CalculatorXamarin/CalculatorXamarin/SimpleCalculator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;

namespace CalculatorXamarin
{
    public class SimpleCalculator : Calculator
    {

        double inputValue1 = 0;
        double inputValue2 = 0;
        Activity Activitycontext;
        public static void Make(Activity context, CalculatorElements elements)
        {

            new SimpleCalculator(context, elements);
        }
        private SimpleCalculator(Activity context, CalculatorElements elements) : base(elements)
        {

            Activitycontext = context;
            AddListeners();
         
        }
        private void AddListeners()
        {
            addOnClearListener();
            addOnAdditionListener();
            addOnSubtractionListener();
            addOnDivisionListener();
            addOnMultiplyListener();
           
        }

        private void addOnClearListener()
        {
            getClearButton().Click += delegate
            {
                Clear();
            };
        }

        private void addOnMultiplyListener()
        {
            getMultiplicationButton().Click += delegate
            {

                if (OperandInputsAreValid())
                {
                    getValuesFromOperandInputs();
                    Multiply();
                }
                else
                {
                    Toast.MakeText(Activitycontext, "please enter  numbers in both operand fields", ToastLength.Long).Show();
                }

            };
        }

        private void addOnDivisionListener()
        {
            getDivisionButton().Click += delegate
            {

                if (OperandInputsAreValid())
                {
                    getValuesFromOperandInputs();
                    Divide();
                }
                else
                {
                    Toast.MakeText(Activitycontext, "please enter  numbers in both operand fields", ToastLength.Long).Show();
                }

            };
        }

        private void addOnSubtractionListener()
        {
            getSubtractionButton().Click += delegate
            {

                if (OperandInputsAreValid())
                {
                    getValuesFromOperandInputs();
                    SubTract();
                }
                else
                {
                    Toast.MakeText(Activitycontext, "please enter  numbers in both operand fields", ToastLength.Long).Show();
                }

            };
        }

        private void addOnAdditionListener()
        {
            getAdditionButton().Click += delegate
            {

                if (OperandInputsAreValid())
                {
                    getValuesFromOperandInputs();
                    Add();
                }
                else
                {
                    Toast.MakeText(Activitycontext, "please enter  numbers in both operand fields", ToastLength.Long).Show();
                }

            };
        }

        public override void Add()
        {
            double result = inputValue1 + inputValue2;
            getResultTxt().Text = result.ToString();

            
        }

        public override void SubTract()
        {
            double result = inputValue1 - inputValue2;
            getResultTxt().Text = result.ToString();
        }

        public override void Divide()
        {
            double result = inputValue1 / inputValue2;
            getResultTxt().Text = result.ToString();
        }

        public override void Multiply()
        {
            double result = inputValue1 - inputValue2;
            getResultTxt().Text = result.ToString();
        }

        public override void Clear()
        {
            getOperand1Txt().Text="";
            getOperand2Txt().Text="";
            getResultTxt().Text="0.00";
            getOperand1Txt().RequestFocus();
        }


        private bool OperandInputsAreValid()
        {
            var isValid = getOperand1Txt().Text.Length > 0 && getOperand2Txt().Text.Length > 0;
            return isValid;
        }

        private void getValuesFromOperandInputs()
        {
            inputValue1 = double.Parse(getOperand1Txt().Text);
            inputValue2 = double.Parse(getOperand2Txt().Text);
        }
    }
}

File: CalculatorAndroidStudio/app/src/main/java/com/practices/samuel/calculatorandroidstudio/SimpleCalculator.java
package com.practices.samuel.calculatorandroidstudio;

import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.Toast;

/**
 * Created by samuel on 7/12/16.
 */
public class SimpleCalculator extends Calculator {

    double inputValue1 = 0;
    double inputValue2 = 0;
    AppCompatActivity Activitycontext;

    private SimpleCalculator(AppCompatActivity context,CalculatorElements elements) {
        super(elements);
        Activitycontext = context;
        addListeners();
    }

    private void addListeners() {
        addOnAdditionListener();
        addOnSubtractionListener();
        addOnDivisionListener();
        addOnMultiplyListener();
        addOnClearListener();
    }

    public static void Make(AppCompatActivity context,CalculatorElements elements){

        new SimpleCalculator(context,elements);
    }

    private void addOnAdditionListener() {
        getAddition().setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                if (OperandInputsAreValid()) {
                    getValuesFromOperandInputs();
                    Add();
                } else {

                    Toast.makeText(Activitycontext, "please enter  numbers in both operand fields", Toast.LENGTH_LONG).show();
                }
            }

        });

    }

    private void addOnSubtractionListener() {
        getSubtraction().setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (OperandInputsAreValid()) {
                    getValuesFromOperandInputs();
                    SubTract();

                } else {

                    Toast.makeText(Activitycontext, "please enter  numbers in both operand fields", Toast.LENGTH_LONG).show();
                }
            }
        });

    }

    private void addOnDivisionListener() {
        getDivision().setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (OperandInputsAreValid()) {
                    getValuesFromOperandInputs();
                    Divide();
                } else {

                    Toast.makeText(Activitycontext, "please enter  numbers in both operand fields", Toast.LENGTH_LONG).show();
                }
            }
        });

    }

    private void addOnMultiplyListener() {
        getMultiplication().setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (OperandInputsAreValid()) {
                    getValuesFromOperandInputs();
                    Multiply();

                } else {

                    Toast.makeText(Activitycontext, "please enter  numbers in both operand fields", Toast.LENGTH_LONG).show();
                }
            }
        });

    }

    private void addOnClearListener() {
        getClear().setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Clear();

            }
        });

    }

    private boolean OperandInputsAreValid() {
        return getOperand1().getText().length() > 0 && getOperand2().getText().length() > 0;
    }

    private void getValuesFromOperandInputs() {
        inputValue1 = Double.parseDouble(getOperand1().getText().toString());
        inputValue2 = Double.parseDouble(getOperand2().getText().toString());
    }
    @Override
    public void Add() {
        double result = inputValue1 + inputValue2;
        getResult().setText(Double.toString(result));

    }

    @Override
    public void SubTract() {
        double result = inputValue1 - inputValue2;
        getResult().setText(Double.toString(result));
    }

    @Override
    public void Divide() {
        double result = inputValue1 / inputValue2;
        getResult().setText(Double.toString(result));
    }

    @Override
    public void Multiply() {
        double result = inputValue1 * inputValue2;
        getResult().setText(Double.toString(result));
    }

    @Override
    public void Clear() {

        getOperand1().setText("");
        getOperand2().setText("");
        getResult().setText("0.00");
        getOperand1().requestFocus();
    }


}
